### PR TITLE
Add parsing xml to robot & download .robot file button

### LINKED ISSUE #9: Implement parsing to .robot
As an RPA developer, I'd like to be able to export a .robot file, in order to run the bot on my local machine. Therefore the diagram needs to be parsed into a .robot file.

---

### :thought_balloon: Hints & Thoughts
No parameter passing yet, only hard coded values. Should only concern the excel task. Have a look [here](https://rpaframework.org/libraries/excel_application/index.html)
Simple XML to .robot parsing.
### :question: Additional Questions to Answer / Consider

---

## :white_check_mark: Conditions of satisfaction

- [ ] .robot File can be run in robotcorp environment
- [ ] Bot achieves what is stated in the BPMN diagram

### LINKED ISSUE #10: Add button to download .robot file
As an RPA Developer, I'd like to be able to download the .robot file of my diagram to execute it on my local machine

---

### :thought_balloon: Hints & Thoughts
Do #9 first
### :question: Additional Questions to Answer / Consider

---

## :white_check_mark: Conditions of satisfaction

- [ ] Added Button on the modeler site
- [ ] Button downloads parsed .robot file
- [ ] File is executable

### PR DESCRIPTION
**Pull request by WolfgangDaniel**
Fixes #9 
Fixes #10 

## Changes
**What does this PR change? What was the problem? What is the solution?**
This PR introduces a parser which parses the underlying xml file of the bpmn diagram into an executable .robot file. Therfore it is now possible to create a robot with a bpmn diagram. Moreover a button was created to trigger the parsing and download of the .robot file. The parser is fully implemented on the client side. 

*describe problem here*
- The Problem is that so far we can only create bpmn diagrams but we cannot create executable .robot files.

## Screenshots
grafik
grafik
grafik

## Checklist before merge

**Developer's responsibilities**
* [ ] Approved by at least one developer: *enter reviewer here*
* [ ] **Pull request build** has passed
* [ ] Production backend has been updated to reflect new features and fix bugs
* [ ] tested locally (in all browsers if frontend)
* [ ] updated the docs
* [ ] added tests where necessary

**Product Owner's responsibilities**
* [ ] No issues have been found on staging environment
* [ ] Product Owner has approved the release

### CODE REVIEW

File: README.md
Comment (kej-jay) [inline README.md:3]: Why has this been removed? Not that I think this is necessary, but I am just curious as to why it was removed.

File: README.md
Comment (lukashueller) [inline README.md:5]: Why have you removed these lines?

File: client/src/components/bpmn.modeler.component.jsx
Comment (lukashueller) [inline client/src/components/bpmn.modeler.component.jsx:57]: Are you sure that a is a speaking variable name?

File: client/src/components/bpmn.modeler.component.jsx
Comment (kej-jay) [inline client/src/components/bpmn.modeler.component.jsx:66]: If possible remove logging before merging

File: client/src/components/bpmn.modeler.component.jsx
Comment (kej-jay) [inline client/src/components/bpmn.modeler.component.jsx:75]: See above comment. I will not comment this anymore, please take care yourself and go through the changes

File: client/src/parser.js
Comment (lukashueller) [inline client/src/parser.js:12]: I would prefer to load this information from our database. What do you think?

File: client/src/components/bpmn.modeler.component.jsx
Comment (kej-jay) [inline client/src/components/bpmn.modeler.component.jsx:77]: Is the filename hardcoded or is there any chance to allow for renaming of the efile before downloading? If it is out of scope for now maybe create a new issue for the future?

File: client/src/parser.js
Comment (lukashueller) [inline client/src/parser.js:23]: I think when pushing we should always remove all ``Console.log()`` statements from the code.

File: client/src/components/bpmn.modeler.component.jsx
Comment (kej-jay) [inline client/src/components/bpmn.modeler.component.jsx:12]: maybe use default export here already? This has been done in other files as well, so I would at least use a consistent scheme. Let me know on what you decide

File: client/src/parser.js
Comment (lukashueller) [inline client/src/parser.js:14]: This looks like a very long function to me. Can we divide this up? Maybe the sections below each comment correspond to a useful function section?

File: client/src/parser.js
Comment (kej-jay) [inline client/src/parser.js:8]: If this is supposed to happen, why did it not happen yet?

File: client/src/parser.js
Comment (kej-jay) [inline client/src/parser.js:3]: There are two functions in this file, but only one documentation for a function. Please put the documentation directly in front of the function which it is supposed to describe and document both functions

File: client/src/parser.js
Comment (kej-jay) [inline client/src/parser.js:14]: Any chance to refactor this function into smaller methods for modularity?

File: client/src/parser.js
Comment (kej-jay) [inline client/src/parser.js:39]: Code should document itself, remove the comment and maybe create a new function to be called with it as the name or shift the comment into the functions description

File: server/index.js
Comment (lukashueller) [inline server/index.js:36]: This code part can be relocated very well in my opinion. Then the ``index.js`` stays clean.

File: client/src/parser.js
Comment (kej-jay) [reply client/src/parser.js:39]: Also the same for the following comments for smaller sections

File: client/src/parser/convertedModelForTesting.json
Comment (kej-jay) [inline client/src/parser/convertedModelForTesting.json:1]: Does this file really belong into the main branch? In my opinion it should be excluded from merging

File: client/src/parser/convertedModelForTesting1Activity.json
Comment (kej-jay) [inline client/src/parser/convertedModelForTesting1Activity.json:1]: IMO this shouldn't be in the repo either necessarily

**Review by lukashueller — CHANGES_REQUESTED** [review]
In principle very good implementation of the desired function. But I miss the documentation for all self-written functions. Additionally, there are some small notes directly on the code. Please use single-line comments only in special cases and Documented functions and classes.

File: client/src/parser/parser.js
Comment (kej-jay) [inline client/src/parser/parser.js:1]: See comment on same file at different location. Please use the comments I commented there and remove one of the files, as I do not think that it is needed twice

**Review by kej-jay — CHANGES_REQUESTED** [review]

File: server/index.js
Comment (kej-jay) [reply server/index.js:36]: IMO it should stay in the index file

File: client/src/parser.js
Comment (kej-jay) [reply client/src/parser.js:12]: I don't even see the benefit of this method... Why not simply `return 'RPA.' + app`

File: README.md
Comment (WolfgangDaniel) [reply README.md:5]: We don't know. I will put them back in.

File: client/src/components/bpmn.modeler.component.jsx
Comment (WolfgangDaniel) [reply client/src/components/bpmn.modeler.component.jsx:57]: Good point. I just replaced a with element :)

File: README.md
Comment (WolfgangDaniel) [reply README.md:3]: We don't know.  I just put them back in.

File: client/src/components/bpmn.modeler.component.jsx
Comment (WolfgangDaniel) [reply client/src/components/bpmn.modeler.component.jsx:66]: Fair point. I just did.

File: client/src/components/bpmn.modeler.component.jsx
Comment (WolfgangDaniel) [reply client/src/components/bpmn.modeler.component.jsx:77]: It is hardcoded and I think renaming is out of scope for now. Sandro just created a new Issue for that.

Comment (WolfgangDaniel) [reply client/src/components/bpmn.modeler.component.jsx:77]: Good point though.

File: server/index.js
Comment (WolfgangDaniel) [reply server/index.js:36]: Okay then I will leave it in the index file

File: client/src/parser/convertedModelForTesting.json
Comment (WolfgangDaniel) [reply client/src/parser/convertedModelForTesting.json:1]: We actually don't need it. I just deleted it.

File: client/src/parser/convertedModelForTesting1Activity.json
Comment (WolfgangDaniel) [reply client/src/parser/convertedModelForTesting1Activity.json:1]: Same here. I just deleted it.

File: client/src/parser/parser.js
Comment (WolfgangDaniel) [reply client/src/parser/parser.js:1]: This parser is obsolete and I removed it.

File: client/src/components/bpmn.modeler.component.jsx
Comment (WolfgangDaniel) [reply client/src/components/bpmn.modeler.component.jsx:12]: We will use default export always at the end of the file.

File: client/src/components/bpmn.modeler.component.jsx
Comment (WolfgangDaniel) [reply client/src/components/bpmn.modeler.component.jsx:75]: See above :)

File: client/src/parser.js
Comment (SanJSp) [reply client/src/parser.js:8]: not enough time

File: client/src/parser.js
Comment (SanJSp) [reply client/src/parser.js:12]: Currently `return 'RPA.' + app` would suffice. I will fix it.

File: server/index.js
Comment (lukashueller) [reply server/index.js:36]: My thought was to simply outsource it into a function.

**Review by lukashueller — APPROVED** [review]

**Review by kej-jay — APPROVED** [review]

File: client/src/components/bpmnModeler/BpmnModeler.jsx
Comment (lukashueller) [inline client/src/components/bpmnModeler/BpmnModeler.jsx:12]: Please reorder imports

File: server/index.js
Comment (lukashueller) [inline server/index.js:8]: This var `bodyParser` is declared but it's value ist never read.

**Review by lukashueller — CHANGES_REQUESTED** [review]
Only a few small things, some I fix directly myself quickly.

File: client/src/components/bpmnModeler/BpmnModeler.jsx
Comment (lukashueller) [reply client/src/components/bpmnModeler/BpmnModeler.jsx]: Done

**Review by lukashueller — DISMISSED** [review]

**Review by lukashueller — CHANGES_REQUESTED** [review]
Unfortunately, I have to decline the approval, because the software is not executable locally. This must be corrected together (see MS TEAMS)!

**Comment by kej-jay** [comment]
> 
> 
> Unfortunately, I have to decline the approval, because the software is not executable locally. This must be corrected together (see MS TEAMS)!

It worked for me. Simply run 'npm install' inside the client folder. from then on everything worked fine for me.

**Review by lukashueller — APPROVED** [review]
After deleting the node modules and reinstalling all dependencies it now works.

**Pull request merged**
